model/task/task.go

```diff
@@ -2036,12 +2028,25 @@ func (t *Task) GetDisplayStatus() string {
 		}
 		return evergreen.TaskSystemFailed
 	}
-	if t.Details.Type == evergreen.CommandTypeSetup {
-		return evergreen.TaskSetupFailed
-	}
 	if t.Details.TimedOut {
 		return evergreen.TaskTimedOut
 	}
+	if t.Status == evergreen.TaskUndispatched {
+		if !t.Activated {
+			return evergreen.TaskUnscheduled
+		}
+		if t.Blocked() {
+			return evergreen.TaskStatusBlocked
+		}
+		return evergreen.TaskWillRun
+	}
+	if !t.IsFinished() {
+		return t.Status
+	}
+	annotation, err := annotations.FindOneByTaskIdAndExecution(t.Id, t.Execution)
```

Comment:
Do we call getDisplayStatus in a lot of places? I'm a little wary about having a DB call here

Reply:
We do call it often but this is a little bit tricky since looking at the task annotations collection is necessary to determine if the display status for a failed task is a "known issue".  Couple things that come to my mind: 

1) A large set of tasks will be caught in the above case and won't make it to this call, and since we now set `DisplayStatus` here it could also decrease the number of invocations we make here since we return early if `DisplayStatus` is set. 

2) We can remove this entirely and accept that in the case of known issue failures, there is a slight deviation between `displayStatusExpression` and `GetDisplayStatus`

Reply:
This concerns me as well. Especially if called from a graphql resolver this can easily spiral into thousands of calls. Could you ensure we always perform a db query that returns a displayStatus there?